File: UnboundApp/UnboundApp/PIXAboutWindowController.m
//
//  PIXAboutWindowController.m
//  UnboundApp
//
//  Created by Scott Sykora on 4/13/13.
//  Copyright (c) 2013 Pixite Apps LLC. All rights reserved.
//

#import "PIXAboutWindowController.h"

@interface PIXAboutWindowController ()

@property (retain) IBOutlet NSTextField * titleField;

@end

@implementation PIXAboutWindowController

- (id)initWithWindow:(NSWindow *)window
{
    self = [super initWithWindow:window];
    if (self) {
        // Initialization code here.
    }
    
    return self;
}

- (void)windowDidLoad
{
    [super windowDidLoad];
    
    NSString *version =[[[NSBundle mainBundle] infoDictionary] valueForKey:@"CFBundleVersion"];
    
    self.titleField.stringValue = [NSString stringWithFormat:@"Unbound v. %@", version];
    
    // Implement this method to handle any initialization after your window controller's window has been loaded from its nib file.
}

-(IBAction)unboundLearnMore:(id)sender
{
    NSURL * url = [NSURL URLWithString:@"http://www.pixiteapps.com"];
    [[NSWorkspace sharedWorkspace] openURL:url];
}

@end

File: UnboundApp/UnboundApp/Windows/PIXAboutWindowController.swift
//
// Created by Ryan Harter on 10/6/16.
// Copyright (c) 2016 Pixite Apps LLC. All rights reserved.
//

import Foundation
import AppKit

class PIXAboutWindowController: NSWindowController {

    @IBOutlet weak var icon: NSImageView!
    @IBOutlet weak var title: NSTextField!
    @IBOutlet weak var version: NSTextField!
    @IBOutlet weak var copyright: NSTextField!
    @IBOutlet weak var link: NSButton!
    @IBOutlet weak var logo: NSImageView!

    private var url: String = ""

    convenience init() {
        self.init(windowNibName: "PIXAboutWindowController")
    }

    override func windowDidLoad() {
        super.windowDidLoad()

        icon.image = NSApp.applicationIconImage

        let infoDict = NSBundle.mainBundle().infoDictionary!
        title.stringValue = infoDict["CFBundleName"] as! String

        let versionNumber = infoDict["CFBundleShortVersionString"] as! String
        let buildNumber = infoDict["CFBundleVersion"] as! String
        version.stringValue = "Version \(versionNumber) (\(buildNumber))"

        copyright.stringValue = infoDict["PIXCopyright"] as! String

        url = infoDict["PIXAppLink"] as! String

        let center = NSMutableParagraphStyle()
        center.alignment = NSCenterTextAlignment
        let attributes = [
                NSUnderlineStyleAttributeName: NSUnderlineStyle.StyleSingle.rawValue,
                NSForegroundColorAttributeName: NSColor.blueColor(),
                NSParagraphStyleAttributeName: center
        ]
        link.attributedTitle = NSAttributedString(string: url, attributes: attributes)
    }

    @IBAction func onAppLinkClicked(sender: AnyObject) {
        NSWorkspace.sharedWorkspace().openURL(NSURL(string: url)!)
    }

    override func keyDown(theEvent: NSEvent) {
        if theEvent.modifierFlags.contains(.CommandKeyMask) && theEvent.charactersIgnoringModifiers! == "w" {
            self.window?.close()
            return
        }
        super.keyUp(theEvent)
    }

}